h2o-py/h2o/automl/_estimator.py

```diff
@@ -404,11 +411,45 @@ def __validate_algo_parameters(self, algo_parameters):
     def __validate_frame(self, fr, name=None, required=False):
         return H2OFrame._validate(fr, name, required=required)
 
+    def __validate_distribution(self, distribution):
+        if is_type(distribution, str):
+            distribution = distribution.lower()
+            if distribution == "custom":
+                raise H2OValueError('Distribution "custom" has to be specified as a '
+                                    'dictionary with their respective parameters, e.g., '
+                                    '`dict(distribution = \"custom\", custom_distribution_func = \"...\"))`.')
+            return distribution
+        if is_type(distribution, dict):
+            dist = distribution["distribution"].lower()
+            ALLOWED_DISTRIBUTION_PARAMETERS = dict(
+                custom='custom_distribution_func',
+                huber='huber_alpha',
+                quantile='quantile_alpha',
+                tweedie='tweedie_power'
+            )
+            assert distribution.get(ALLOWED_DISTRIBUTION_PARAMETERS.get(dist)) is not None or len(distribution) == 1, (
+                    "Distribution dictionary should contain distribution and a distribution "
+                    "parameter. For example `dict(distribution=\"{}\", {}=...)`."
+            ).format(dist, ALLOWED_DISTRIBUTION_PARAMETERS[dist])
+            if distribution["distribution"] == "custom" and "custom_distribution_func" not in distribution.keys():
+                raise H2OValueError('Distribution "custom" has to be specified as a '
+                            'dictionary with their respective parameters, e.g., '
+                            '`dict(distribution = \"custom\", custom_distribution_func = \"...\"))`.')
+            if ALLOWED_DISTRIBUTION_PARAMETERS.get(dist) in distribution.keys():
+                setattr(self, ALLOWED_DISTRIBUTION_PARAMETERS[dist], distribution[ALLOWED_DISTRIBUTION_PARAMETERS[dist]])
+            return dist
+
+
     _extract_params_doc(getdoc(__init__))
     project_name = _aml_property('build_control.project_name', types=(None, str), freezable=True,
                                  validate_fn=__validate_project_name)
     nfolds = _aml_property('build_control.nfolds', types=(int,), freezable=True,
                            validate_fn=__validate_nfolds)
+    distribution = _aml_property('build_control.distribution', types=(str, dict), freezable=True, validate_fn=__validate_distribution)
+    custom_distribution_func = _aml_property('build_control.custom_distribution_func', types=(str,), freezable=True)
+    huber_alpha = _aml_property('build_control.huber_alpha', types=(numeric,), freezable=True)
+    tweedie_power = _aml_property('build_control.tweedie_power', types=(numeric,), freezable=True)
+    quantile_alpha = _aml_property('build_control.quantile_alpha', types=(numeric,), freezable=True)
```

Comment:
can you make those properties "private" (prefixed with underscore)?
Otherwise it provides 2 official ways to set those properties: the dict keys on `distribution` property, or directly through those properties, except that there won't be proper validation in the latter case.

Reply:
Done. Thank you for the explanation!